pkg/certificates/certificate_subject.go

```diff
@@ -29,3 +36,45 @@ func (c certificateSubjects) createCertificateSubjects() []certificateSubject {
 	return subjects
 
 }
+
+func (c certificateSubject) generateCertificateTemplate(alternativeIP string) (x509.Certificate, *rsa.PrivateKey, error) {
```

Comment:
I would leave this as an function, not method. You just use subjectIP and don't change anything in subject struct.

Reply:
Why? I did not change subject struct but I don't see a point in having this as a function not a method. Creation of certificate is for subject and it uses it inners